temporian/implementation/numpy/operators/equal.py

```diff
@@ -0,0 +1,81 @@
+from typing import Dict
+
+import numpy as np
+
+from temporian.core.operators.equal import EqualOperator
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.feature import NumpyFeature
+from temporian.implementation.numpy.data.feature import DTYPE_REVERSE_MAPPING
+from temporian.implementation.numpy import implementation_lib
+from temporian.implementation.numpy.operators.base import OperatorImplementation
+
+
+class EqualNumpyImplementation(OperatorImplementation):
+    """New event with booleans where features are equal to a value."""
+
+    def __init__(self, operator: EqualOperator) -> None:
+        super().__init__(operator)
+
+    def __call__(self, event: NumpyEvent) -> Dict[str, NumpyEvent]:
+        def are_same_general_type(dtype1, dtype2) -> bool:
```

Comment:
Could this check be made in the operator definition (!=operator implementation) in such a way that an error is raised if the two types are not compatible (instead of always returning false)?